codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/serde/CborSerializerGenerator.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.smithy.kotlin.codegen.rendering.serde
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.codegen.core.SymbolReference
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.RuntimeTypes
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import software.amazon.smithy.kotlin.codegen.model.knowledge.SerdeIndex
+import software.amazon.smithy.kotlin.codegen.model.targetOrSelf
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.ProtocolGenerator
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.toRenderingContext
+import software.amazon.smithy.model.shapes.*
+import software.amazon.smithy.model.traits.TimestampFormatTrait
+
+class CborSerializerGenerator(
+    private val protocolGenerator: ProtocolGenerator,
+) : StructuredDataSerializerGenerator {
+    override fun operationSerializer(ctx: ProtocolGenerator.GenerationContext, op: OperationShape, members: List<MemberShape>): Symbol {
+        val input = op.input.get().let { ctx.model.expectShape(it) }
+        val symbol = ctx.symbolProvider.toSymbol(input)
+
+        return op.bodySerializer(ctx.settings) { writer ->
+            addNestedDocumentSerializers(ctx, op, writer)
+            writer.withBlock("private fun #L(context: #T, input: #T): ByteArray {", "}", op.bodySerializerName(), RuntimeTypes.Core.ExecutionContext, symbol) {
+                call {
+                    renderSerializeOperationBody(ctx, op, members, writer)
+                }
+            }
+        }
+    }
+
+    private fun renderSerializeOperationBody(
+        ctx: ProtocolGenerator.GenerationContext,
+        op: OperationShape,
+        documentMembers: List<MemberShape>,
+        writer: KotlinWriter,
+    ) {
+        val shape = ctx.model.expectShape(op.input.get())
+        writer.write("val serializer = #T()", RuntimeTypes.Serde.SerdeCbor.CborSerializer)
+        renderSerializerBody(ctx, shape, documentMembers, writer)
+        writer.write("return serializer.toByteArray()")
+    }
```

Comment:
I thought this would "just work", but there is one issue with this change. `SdkSource.toHttpBody` creates a non-replayable body, which means we can't calculate the hash during signing (specifically, canonicalization). 

I believe we will still need to read this into a byte array to ensure it's replayable. [`764fea6` (#1103)](https://github.com/smithy-lang/smithy-kotlin/pull/1103/commits/764fea66262780a6b3c8075ab5809d651f748555)